velox/exec/tests/utils/PlanBuilder.h

```diff
@@ -464,6 +464,11 @@ class PlanBuilder {
       const std::vector<std::string>& aggregationInputs,
       std::string groupIdName = "group_id");
 
+  ///  Add a ExpandNode to apply all of the projections expressions to every
```

Comment:
It is difficult to explain what this plan node does, so let's refer to ExpandNode's documentation instead.

```
Add an ExpandNode using specified projections. See comments for ExpandNode class for description of this plan node.
```

Let's clarify how 'names' are generated. I assume these are generated using the first list of projections and default to p0, p1, p2 unless projection SQL includes an alias. Please, clarify.

Reply:
@mbasmanova  Assuming the input data type is `k1, k2, a, b`, and the expand projections are as follows:
```
{"k1", "k2", "a", "b", "0"},
{"k1", "", "a", "b", "1"},
{"", "k2", "a", "b", "2"},
{"", "", "a", "b", "3"}
```
In each projection, the item represents the name for each column except the constant value column. To determine the name of each column, we traverse each projection and follow these rules: If the column corresponding to the current projection is not empty, we will take the current value as the name. For the constant value column, we generate the name as "group_id_" followed by the count of constant value columns. The resulting column names are: [k1, k2, a, b, group_id0].

Reply:
@JkSelf Does Spark use Expand operator to implement 'LATERAL VIEW EXPLODE' HQL? If so, let's make sure Expand operator and related utilities are defined in generic terms not specific to grouping sets. Therefore, let's not use group_idN naming here.

https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-lateral-view.html

Reply:
@mbasmanova No. Spark will generate `GenerateExec  `operator for 'LATERAL VIEW EXPLODE' HQL not Expand.

Reply:
@JkSelf GenerateExec sounds very generic. What does this operator do? How do you map it to Velox? Any chance you could share some query plans for queries with grouping sets and LATERAL VIEW EXPLODE ?

Reply:
@mbasmanova The [GenerateExec](https://github.com/apache/spark/blob/1fdd46f173f7bc90e0523eb0a2d5e8e27e990102/sql/core/src/main/scala/org/apache/spark/sql/execution/GenerateExec.scala#L42) operator applies a generator expression to a stream of input rows, where the generator expression can produce zero or more rows based on a single input row. Generate + Explode can be implemented using velox's Unnest. We are developing this feature in Gluten and the PR link is [here](https://github.com/oap-project/gluten/pull/3068).

**Explode sql and query plan:**
```
select explode(array(10, 20))

== Physical Plan ==
*(1) Generate explode(10， 20), false, [col#0]
   +- *(1) Scan OneRowRelation [codegen id : 1]

```

**Groupingsets sql and query plan:**

```
select sum(l_suppkey) from lineitem where group by ROLLUP(l_orderkey, l_partkey)

== Physical Plan ==\n
AdaptiveSparkPlan isFinalPlan=false\n
+- HashAggregate(keys=[l_orderkey#42L, l_partkey#43L, spark_grouping_id#41L], functions=[sum(l_suppkey#2L)])\n
   +- Exchange hashpartitioning(l_orderkey#42L, l_partkey#43L, spark_grouping_id#41L, 200), ENSURE_REQUIREMENTS, [plan_id=26]\n
      +- HashAggregate(keys=[l_orderkey#42L, l_partkey#43L, spark_grouping_id#41L], functions=[partial_sum(l_suppkey#2L)])\n
         +- Expand [[l_suppkey#2L, l_orderkey#0L, l_partkey#1L, 0], [l_suppkey#2L, l_orderkey#0L, null, 1], [l_suppkey#2L, null, null, 3]], [l_suppkey#2L, l_orderkey#42L, l_partkey#43L, spark_grouping_id#41L]\n
            +- BatchScan[l_orderkey#0L, l_partkey#1L, l_suppkey#2L] ParquetScan DataFilters: [], Format: parquet, Location: InMemoryFileIndex(1 paths)[file:/home/sparkuser/jk/projects/gluten-3.2/gluten-core/src/test/resou..., PartitionFilters: [], PushedFilters: [], ReadSchema: struct<l_orderkey:bigint,l_partkey:bigint,l_suppkey:bigint>, PushedFilters: [] RuntimeFilters: []\n\n|

```

Reply:
@JkSelf Thank you. This is helpful.